lib/utils.js

```diff
@@ -7,98 +10,79 @@ var assert = require("assert");
  * @returns Configuration Object
  */
 exports.parseMongoConnectionURL = function (mongoConnectionURL) {
```

Comment:
I rewrote most of this, but I feel it makes it much easier to digest. Instead of needing it follow string parsing logic you now just need to realise we use `url.parse` on each section of the connection string. It's also broken down into three small functions now:

* `getName`
* `getAuthenticationObject`
* `getOptionsObject`